YellowNotes.Api/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using YellowNotes.Api.Extensions;
+using YellowNotes.Core.Dtos;
+using YellowNotes.Core.Repositories;
+
+namespace YellowNotes.Api.Controllers
+{
+    [ApiController]
+    [Route("/[controller]")]
+    public class CategoriesController : ControllerBase
+    {
+        private readonly ICategoryRepository categoryRepository;
+
+        public CategoriesController(ICategoryRepository categoryRepository) =>
+            this.categoryRepository = categoryRepository;
+
+        [HttpGet]
+        [ProducesResponseType(typeof(IEnumerable<CategoryDto>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<IEnumerable<CategoryDto>>> GetCategories(
```

Comment:
should be always as generic as possible, in your particular case should be 'IActionResult'.